plenum/test/replica/test_process_prepare.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from plenum.common.exceptions import SuspiciousNode
+from plenum.common.util import updateNamedTuple
+from plenum.server.suspicion_codes import Suspicions
+from plenum.test.helper import generate_state_root
+
+nodeCount = 7
+
+NON_PRIMARY_NAME = "Gamma:0"
+
+
+@pytest.fixture(scope='function', params=[0])
+def viewNo(tconf, request):
+    return request.param
+
+
+@pytest.fixture(scope="function",
+                params=['BLS_not_None'])
+def multi_sig(fake_multi_sig, request):
+    if request.param == 'BLS_None':
+        return None
+    return fake_multi_sig
+
+
+@pytest.fixture(scope='function', params=['Primary', 'Non-Primary'])
+def r(replica_with_requests, request):
```

Comment:
I would prefer more self-explanatory name like `replica` or `test_replica` or something else, but not one-letter `r`

Reply:
Me too, but `replica` doesn't work as it is because of a parent fixture with the same name. It's possible to solve this, but may lead to non-obvious imports and code.